Example/MatrioskaExample/AppDelegate.swift

```diff
@@ -14,71 +14,60 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
-                        ),
-                        tileComponent(meta: TileConfig(text: "Three", color: .orange)),
-                        tileComponent(meta: TileConfig(text: "Four", color: .yellow)),
-                    ],
-                    meta: ZipMeta(ClusterLayout.TabConfig(title: "Third", iconName: "tabIcon"))
-                )
-            ),
-            blankComponent(
-                meta: ClusterLayout.TabConfig(title: "Fourth", iconName: "tabIcon")
-            ),
-            blankComponent(
-                meta: ClusterLayout.TabConfig(title: "Fifth", iconName: "tabIcon")
-            ),
-        ],
-        meta: ClusterLayout.TabBarConfig(selectedIndex: 2)
-    )
+    /**/
 
-    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+    func application(_ application: UIApplication,
+                     didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
         
         window = UIWindow(frame: UIScreen.main.bounds)
-        window?.rootViewController = rootComponent.viewController()
-        window?.makeKeyAndVisible()
+        
+        let factory = JSONFactory()
+        
+        let tabBarBuilder: JSONFactory.ClusterFactoryBuilder = { (children, meta) in
+            ClusterLayout.tabBar(children: children, meta: meta)
+        }
+        
+        let stackBuilder: JSONFactory.ClusterFactoryBuilder = { (children, meta) in
+            ClusterLayout.stack(children: children, meta: meta)
+        }
+        
+        let navigationBuilder: JSONFactory.WrapperFactoryBuilder = { (child, meta) in
+            Component.wrapper(builder: { (child, meta) in
+                guard let vc = child.viewController() else {
+                    return nil
+                }
+                return UINavigationController(rootViewController: vc)
+            }, child: child, meta: meta)
+        }
+        
+        let tileBuilder: JSONFactory.ViewFactoryBuilder = { (meta) in
+            Component.view(builder: { (meta) -> UIViewController? in
+                    TileViewController.init(meta: TileConfig.materialize(meta))
+                }, meta: meta)
+        }
+        
+        let matrioskaCodeBuilder: JSONFactory.ViewFactoryBuilder = { (meta) in
+            Component.view(builder: MatrioskaCodeViewController.init(meta:), meta: meta)
+        }
+        
+        factory.register(with: "tabbar", factoryBuilder: tabBarBuilder)
+        factory.register(with: "stack", factoryBuilder: stackBuilder)
+        factory.register(with: "navigation", factoryBuilder: navigationBuilder)
+        factory.register(with: "tile", factoryBuilder: tileBuilder)
+        factory.register(with: "matrioska", factoryBuilder: matrioskaCodeBuilder)
+        
+        do {
+            if let json = try JSONReader.jsonObject(from: "app_structure"),
```

Comment:
JSONFactory should take care of `app_structure` I guess, will update that with the schema documentation.